src/app.rs

```diff
@@ -183,25 +183,26 @@ impl Application {
 
         let mut config_paths = self.config.config_paths;
 
-        #[cfg(feature = "api")]
-        // assigned to prevent the API terminating when falling out of scope
-        let _api = if self.config.api.enabled {
-            emit!(ApiStarted {
-                addr: self.config.api.bind.unwrap(),
-                playground: self.config.api.playground
-            });
-
-            Some(api::Server::start(self.config.api))
-        } else {
-            None
-        };
-
         let opts = self.opts;
+        let api_config = self.config.api;
```

Comment:
Seems like this `let` needs `#[cfg(feature = "api")]` too? Maybe it could be moved into the below block?

Reply:
Thanks, `#[cfg(feature = "api")]` would prevent an unnecessary copy when it's not enabled 👍 

It can't be moved into the block, unfortunately, since that'd take ownership of `self`. The type here is `config::api::Option`, which derives `Copy`, which is why I'm able to assign this despite it looking like something that could be moved.

Reply:
:+1: it stood out because the integration tests failed (presumably because they don't enable this feature so `self.config` had no `api` field).

Reply:
Just noticed that too, thanks!